pkg/service/service.go

```diff
@@ -62,9 +62,13 @@ func NewService(ctx context.Context, connection ctrl.Connection) *service {
 }
 
 func (c *service) SendAndWaitForAck(opcode ctrl.OpCode, payload encoding.BinaryMarshaler) error {
-	b, err := payload.MarshalBinary()
-	if err != nil {
-		return err
+	var b []byte
+	var err error
+	if payload != nil {
```

Comment:
Do we want to restrict these kinda "ping" messages, w/o no payload to certain use-cases?

Generally useful to (eventually) document the protocol

Reply:
I don't know, as is now i catched it while running the conformance tests. Maybe a method dedicated to such "signals" (messages without payload) is better?